shell/server/pre-meteor.js

```diff
@@ -210,7 +210,24 @@ function checkMagic(buf, magic) {
   return true;
 }
 
-function serveStaticAsset(req, res) {
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      return serveStaticAsset(req, res, {
```

Comment:
I dislike this overriding on `serveStaticAsset`.  They are not conceptually related, and I think this is the wrong level of abstraction to reuse code at.  I'd rather you just

```
res.writeHead(200, {"Content-Type": "text/plain"});
res.end("Self-test OK.");
```

for `GET`, extract a method for handling the 'OPTIONS' request (if the proper handling matches the one in `serveStaticAsset` exactly), and explicitly 405 anything else.